### ISSUE TITLE
feature request: "add rule for this site" action

### ISSUE THREAD

**Issue by henk84** [foxyproxy/firefox-extension#200]
Hi

I use foxyproxy with tor and some websites block tor in general.
I’d very much appreciate a shortcut in the addon dropdown menu that allows disabling foxyproxy for this site. Or something like "add rule for this site" that spawns the "add proxy" page of the options with the URL prefilled.

Thank you for considering this and for providing foxyproxy!

**Comment by erosman** [comment]
> menu that allows disabling foxyproxy for this site. 

Let's see if there is a popular demand for it.

> Or something like "add rule for this site" that spawns the "add proxy" page of the options with the URL prefilled

It has already been done for v8.0

Previously: #172

**Comment by henk84** [comment]
> > menu that allows disabling foxyproxy for this site.

> Let's see if there is a popular demand for it.

#172 with the additional options "use system settings" and "direct (no proxy)" (as in the "add" dialog’s "proxy type" list) would be enough to satisfy this (at least for me).

> > Or something like "add rule for this site" that spawns the "add proxy" page of the options with the URL prefilled

> It has already been done for v8.0
> Previously: #172

Very cool!
Is there any way to install v8.0, i.e. take part in beta-testing?
Is there any estimation when v8.0 will be released?

**Comment by erosman** [comment]
> https://github.com/foxyproxy/firefox-extension/issues/172 with the additional options "use system settings" and "direct (no proxy)" (as in the "add" dialog’s "proxy type" list) would be enough to satisfy this (at least for me).

There is no proxy to use with 'system' or 'direct' in v8.0.
However, v8.0 has a global bypass list option.

**Comment by ericjung** [comment]
I can setup a beta for you. Do you use Firefox or chrome?

**Comment by erosman** [comment]
FoxyProxy v8.0 beta is available from [FoxyProxy v8.0](https://github.com/foxyproxy/browser-extension)

ref: #196

**Comment by henk84** [comment]
> FoxyProxy v8.0 beta is available from [FoxyProxy v8.0](https://github.com/foxyproxy/browser-extension)
> 
> ref: #196

I have been to that page and that issue but I think I’m missing something. How would I install it in firefox from there?

**Comment by erosman** [comment]
> I have been to that page and that issue but I think I’m missing something. How would I install it in firefox from there?

Please use https://github.com/foxyproxy/browser-extension/issues for v8.0 issues.

1. Go to https://github.com/foxyproxy/browser-extension
2. Click Code button -> Download Zip
3. Unzip (you only need the `src` folder)
4. Use Firefox Nightly, Beta or Developer Edition
5. Firefox: copy `manifest-firefox.json` as `manifest.json` and leave in the same folder
6. Go to: `about:debugging#/runtime/this-firefox` 
7. Click `Load Temporary Add-on...`
8. Navigate to above `manifest.json`

**Comment by henk84** [comment]
>     4. Use Firefox Nightly, Beta or Developer Edition

Thank you very much for this guide!

Installing it seemed to have worked even on firefox-ESR but changing to "proxy by patterns" in the toolbar icon’s dropdown _seems_ to work but closing and immediately opening it again, "disabled" is selected again.
Setting it statically to one of the configured proxies works, though.

I’m assuming this is due to using ESR and not one of the versions you mentioned.

Neither "the time I can spend on this" nor "my actual need for this feature" are actually very high, so unless you say it should work on 102.6-ESR and want more info from me, I’ll just wait for v8.0 to be released as stable.

AFAICT v8 brings everything I wanted so feel free to close this issue.

Thank you both very much for your efforts!

**Comment by erosman** [comment]
> but closing and immediately opening it again, "disabled" is selected again.

There was a minor oversight when updating the latest UI. 
Fixed.

**Comment by henk84** [comment]
> > #172 with the additional options "use system settings" and "direct (no proxy)" (as in the "add" dialog’s "proxy type" list) would be enough to satisfy this (at least for me).
> 
> There is no proxy to use with 'system' or 'direct' in v8.0. However, v8.0 has a global bypass list option.

That’s cool. And it would be even better if the list in "Add host pattern to" would include "global bypass list". ATM it just lists the configured proxies which does not help me.

To repeat my usecase: my default proxy (tor) is blocked on some sites and I would like an easy way to disable using a proxy for these sites.

**Comment by erosman** [comment]
*Global Exclude* is a new feature in v8.0 (and still under development).
It is possible to add a "Add host pattern to Global Exclude" menu as well.
I will look into it once other issues are sorted.

**Comment by erosman** [comment]
"Exclude Host" feature was added to v8.0

**Comment by henk84** [comment]
> "Exclude Host" feature was added to v8.0

I see the text "exclude host" underneath "add host pattern to" in the icon popup but it’s just text, clicking it does not do anything.
How is it supposed to work?

**Comment by erosman** [comment]
It is still under development. 

Code updated. You can try and let me know.

**Comment by henk84** [comment]
> Code updated. You can try and let me know.

Hi, sorry for the delay.

Seems to work, thank you!

Coming back to my initial motivation for requesting this:
some sites have stuff included that breaks when accessed via a proxy (tor in this case), i.e. not the domain one is actually accessing is giving trouble but a "third party" domain. ("third party" for lack of a better word, because it’s sometimes not actually "third party", but "first party", yet a different domain.)
Sadly, the example I thought I had seems to work fine now, so I can’t give a proper example ATM.

Would it be possible to continue having rules based on domain of the actual request but additionally allow creating rules for some of the following
* all requests made in this tab
* all requests made in the context of loading $domain (loading something like maps.google.com would also cause requests to googleapis.com and I would like to be able to tell foxyproxy to treat all requests in the context of maps.google.com with the same rule)
* all requests made by a given container

Thank you for considering!

**Comment by Hideman85** [comment]
Hi there, I was exactly looking at this, making the proxy by pattern can be complex cause websites uses different domains etc that you would all need to know and bind in settings.
So I was looking for turn this proxy for this tab so then all network of this tab is proxified while rest remain normal or have use of different proxy.
Looking forward to see this option 🙂

**Comment by ericjung** [comment]
The feature originally requested in this ticket is now implemented (previous FoxyProxy versions called it QuickAdd but I see that it’s not called that anymore).

It’s not appropriate to make a different request in the same ticket. please open another ticket or see if your request already exists in another ticket.

**Comment by erosman** [comment]
Please post all the features requests to [FoxyProxy v8.0](https://github.com/foxyproxy/browser-extension).
